### PR TITLE
DealDeviceTwin cleanup

### PR DESCRIPTION
**Pull request by TianTianBigWang**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
needn't check dealType twice, if err != nil return err is ok

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by gy95** [comment]
lgtm

File: edge/pkg/devicetwin/dtmanager/twin.go [hunk 1]
```diff
@@ -207,14 +207,12 @@ func DealDeviceTwin(context *dtcontext.DTContext, deviceID string, eventID strin
...
+	if dealType == RestDealType {
 		updateResult, _ := dttype.BuildDeviceTwinResult(dttype.BaseMessage{EventID: eventID, Timestamp: now}, dealTwinResult.Result, dealType)
 		dealUpdateResult(context, deviceID, eventID, dtcommon.InternalErrorCode, err, updateResult)
-		return err
 	}
-	if dealType == RestDealType {
-		updateResult, _ := dttype.BuildDeviceTwinResult(dttype.BaseMessage{EventID: eventID, Timestamp: now}, dealTwinResult.Result, dealType)
-		dealUpdateResult(context, deviceID, eventID, dtcommon.InternalErrorCode, nil, updateResult)
+	if err != nil {
+		return err
```
Comment (gy95) [inline edge/pkg/devicetwin/dtmanager/twin.go:215]: should be like this: `if err != nil && dealType == RestDealType`
Or logic will change and need more considerations

considering one circumstance: if  dealType==SyncDealType and err != nil, this modification will return directly but will continue to process without this patch.

Comment (TianTianBigWang) [reply edge/pkg/devicetwin/dtmanager/twin.go]: Yes, I thought about it. It does cause a return when `err!=nil && dealType == 1`, without executing subsequent code, so I put `if err != nil` in the `dealType` judgment

**Review by fisherxu — COMMENTED** [review]
/approve

**Pull request merged**